x-pack/plugins/infra/server/lib/alerting/metric_threshold/lib/evaluate_alert.ts

```diff
@@ -35,6 +39,12 @@ interface Aggregation {
   };
 }
 
+type Aggregation = AggregationWithIntervals | AggregationWithoutIntervals;
+
+function isAggregationWithIntervals(subject: any): subject is AggregationWithIntervals {
```

Comment:
can we use `subject?: Aggregation` here? or `subject: Aggregation | undefined` similar to what `aggregations` is below, where this function is called?